src/uucore/src/lib/features/fs.rs

```diff
@@ -743,6 +743,52 @@ pub fn path_ends_with_terminator(path: &Path) -> bool {
         .map_or(false, |wide| wide == b'/'.into() || wide == b'\\'.into())
 }
 
+pub mod sane_blksize {
+
+    #[cfg(not(target_os = "windows"))]
+    use std::os::unix::fs::MetadataExt;
+    use std::{fs::metadata, path::Path};
+
+    pub const DEFAULT: u64 = 512;
+    pub const MAX: u64 = (u32::MAX / 8 + 1) as u64;
+
+    /// Provides sanity checked blksize value from the provided value.
+    /// If the provided value is a invalid values a meaningful adaption
+    /// of that value is done.
+    pub fn sane_blksize(st_blksize: u64) -> u64 {
+        match st_blksize {
+            0 => DEFAULT,
+            1..=MAX => st_blksize,
+            _ => DEFAULT,
+        }
+    }
+
+    /// Provides the blksize information from the provided metadata.
+    /// If the metadata contain invalid values a meaningful adaption
+    /// of that value is done.
+    pub fn sane_blksize_from_st(_st: &std::fs::Metadata) -> u64 {
```

Comment:
What does `st` mean? The function gets the `blksize` from the metadata, so I don't understand why the function isn't named `sane_blksize_from_metadata`.

Reply:
`st` comes from the initial implementation in head where `st` stands for `stat` which refers to the unix function [fstat](https://linux.die.net/man/2/fstat) and struct name that provides the metadata.

But you are right. This context is not obvious. I'll rename it according to your proposal.

Reply:
Thanks for your explanation!